utils/io/fileutils/files.go

```diff
@@ -164,8 +164,36 @@ func ListFilesRecursiveWalkIntoDirSymlink(path string, walkIntoDirSymlink bool)
 	return
 }
 
+func ListFilesWithFilterFunc(rootPath string, isRecursive, includeDirs, walkIntoDirSymlink bool, filterFunc func(filePath string) (bool, error)) ([]string, error) {
+	if isRecursive {
+		return listFilesRecursiveWithFilterFunc(rootPath, walkIntoDirSymlink, filterFunc)
+	} else {
+		return listFilesNonRecursiveWithFilterFunc(rootPath, includeDirs, filterFunc)
+	}
+}
+
+// Return the recursive list of files and directories in the specified path
+func listFilesRecursiveWithFilterFunc(path string, walkIntoDirSymlink bool, filterFunc func(filePath string) (bool, error)) (fileList []string, err error) {
+	fileList = []string{}
+	err = gofrog.Walk(path, func(path string, f os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		satisfy, err := filterFunc(path)
+		if err != nil {
+			return err
+		}
+		if satisfy {
+			fileList = append(fileList, path)
+		}
+		return nil
+	}, walkIntoDirSymlink)
+	err = errorutils.CheckError(err)
+	return
+}
+
 // Return all files in the specified path who satisfy the filter func. Not recursive.
-func ListFilesByFilterFunc(path string, filterFunc func(filePath string) (bool, error)) ([]string, error) {
+func listFilesNonRecursiveWithFilterFunc(path string, includeDirs bool, filterFunc func(filePath string) (bool, error)) ([]string, error) {
```

Comment:
Can we unite listFilesRecursiveWithFilterFunc and listFilesNonRecursiveWithFilterFunc to use only gofrog.Walk?

Reply:
@yahavi How can we use gofrog.Walk not recursive?

Reply:
I'm not sure. Something like this?

```go
	err = gofrog.Walk(rootPath, func(path string, f os.FileInfo, err error) error {
		if err != nil {
			return err
		}
                // ...
		if nonRecursive && rootPath != path {
			return gofrog.ErrSkipDir
		}
		return nil
	}, walkIntoDirSymlink)
```

Reply:
Gaon @yahavi

Reply:
Cancel the Gaon, it was wrong 😞 
the files on the top level are not ==rootPath
they are rootPath/file.
@yahavi